wandb/sdk/lib/server.py

```diff
@@ -3,15 +3,25 @@
 """
 
 import json
+import os
+import sys
 
 from wandb import util
 from wandb.apis import InternalApi
+from wandb.errors.error import CommError
 
 
 class ServerError(Exception):
     pass
 
 
+def _is_kaggle() -> bool:
+    return (
+        os.getenv("KAGGLE_KERNEL_RUN_TYPE") is not None
+        or "kaggle_environments" in sys.modules  # noqa: W503
+    )
```

Comment:
We already have this code in `wandb_settings.py`.  Likely makes sense to put this method in `wandb.util` and share it in both places.